rust/openvasd/src/tls.rs

```diff
@@ -175,24 +172,60 @@ pub fn tls_config(config: &crate::config::Config) -> Result<Option<TlsData>, Err
         Some(x) => x,
         None => return Ok(None),
     };
+
     let mut roots = RootCertStore::empty();
     for root in clients.iter().flat_map(load_certs).flatten() {
         roots.add(root)?;
     }
     let key = load_private_key(&key)?;
     let certs = load_certs(&certs)?;
 
-    let inner = WebPkiClientVerifier::builder(roots.into()).build()?;
-    let client_identifier = Arc::new(RwLock::new(ClientIdentifier::default()));
-    let verifier = ClientSnitch::new(inner, client_identifier.clone()).boxed();
+    if !roots.is_empty() {
+        let inner = WebPkiClientVerifier::builder(roots.into()).build()?;
+        let client_identifier = Arc::new(RwLock::new(ClientIdentifier::default()));
+        let verifier = ClientSnitch::new(inner, client_identifier.clone()).boxed();
+
+        let mut config: ServerConfig = ServerConfig::builder()
+            .with_client_cert_verifier(verifier)
+            .with_single_cert(certs, key)?;
 
-    let mut config: ServerConfig = ServerConfig::builder()
-        .with_client_cert_verifier(verifier)
-        .with_single_cert(certs, key)?;
+        config.alpn_protocols = vec![b"h2".to_vec()];
 
-    config.alpn_protocols = vec![b"h2".to_vec()];
+        Ok(Some((client_identifier, config, !clients.is_empty())))
+    } else {
+        match &config.tls.client_certs {
+            Some(clicerts) => {
+                match std::fs::read_dir(clicerts) {
+                    Ok(_x) => {
```

Comment:
```suggestion
                    Ok(_) => {
```
I would rather remove the x completely, or is there a reason for this?

Reply:
it was the clippy's suggestion.